### PR TITLE
adding in use effect and page tracking from onFetch

### PR DESCRIPTION
**Pull request by LoganArnett**
This PR is for ticket [#169606899](https://www.pivotaltracker.com/story/show/169606899)

s

### CODE REVIEW

File: app/javascript/components/lease_ups/LeaseUpApplicationsTable.js [hunk 1]
```diff
@@ -107,14 +113,18 @@ const LeaseUpApplicationsTable = ({ listingId, dataSet, onLeaseUpStatusChange, o
...
       pages={pages}
       columns={columns}
       getTdProps={getTdProps}
       getTrProps={getTrProps}
       defaultPageSize={rowsPerPage}
       sortable={false}
       loading={loading}
-      onFetchData={onFetchData}
+      onFetchData={(state, instance) => {
+        setCurrentPage(state.page)
```
Comment (akegan) [inline app/javascript/components/lease_ups/LeaseUpApplicationsTable.js:125]: can you explain a bit what's happening here? ReactTable has its own page in its state, right, so here we're saying, to update the react table's state to have the current page?

I found this answer for our issue, is that where you got the idea? https://spectrum.chat/react-table/general/pagination-reset-on-data-changed~2ef35446-d56a-4098-ac11-3e344f3af085

Comment (LoganArnett) [reply app/javascript/components/lease_ups/LeaseUpApplicationsTable.js:125]: Similar solution yes, I didn't actually see this prior, I read the SSR docs for the `manual` flag that we have set. We keep track of the page whenever data changes in the `onFetchData` call so we can set it by default with `useState` and then track it in state directly when the onFetch function fires

**Comment by LoganArnett** [comment]
@akegan the update you made earlier allows this PR to pass

**Review by akegan — APPROVED** [review]
cool, looks good

**Pull request merged**